### PR TITLE
ddl, model: clean up

### PR DESCRIPTION
**Pull request by zimulala**
1. Rename updateTableInfo to updateVersionAndTableInfo.
2. We maybe forget to assign some values to the DDL job when we finish it. So I add the `Finish` function to make us remember to handle it.

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @coocood @winkyao @shenli

File: ddl/table.go [hunk 1]
```diff
@@ -51,7 +51,6 @@ func (d *ddl) onCreateTable(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	switch tbInfo.State {
 	case model.StateNone:
 		// none -> public
-		job.SchemaState = model.StatePublic
```
Comment (shenli) [inline ddl/table.go]: Why remove this?

File: ddl/table.go [hunk 2]
```diff
@@ -367,9 +358,10 @@ func checkTableNotExists(t *meta.Meta, job *model.Job, schemaID int64, tableName
 	return nil
 }
 
-func updateTableInfo(t *meta.Meta, job *model.Job, tblInfo *model.TableInfo, originalState model.SchemaState) (
+// updateVersionAndTableInfo updates the schema version and the table information.
+func updateVerionAndTableInfo(t *meta.Meta, job *model.Job, tblInfo *model.TableInfo, shouldUpdateVer bool) (
```
Comment (shenli) [inline ddl/table.go]: updateVerionAndTableInfo  -> updateVersionAndTableInfo

File: ddl/table.go (hunk 1 above)
Comment (zimulala) [reply ddl/table.go]: I don't remove it, it is handled in `Finish`.

File: model/ddl.go [hunk 3]
```diff
@@ -144,6 +144,20 @@ type Job struct {
 	Version int64 `json:"version"`
 }
 
+// Finish is called when a job is finished.
+// It updates the job's state information and adds the binlog information.
+// If dbInfo isn't nil, we add dbInfo to binlog.
+// If dbInfo is nil, we add tblInfo to binlog.
+func (job *Job) Finish(jobState JobState, schemaState SchemaState, ver int64, dbInfo *DBInfo, tblInfo *TableInfo) {
```
Comment (coocood) [inline model/ddl.go]: How about define two functions, one for TableInfo, the other for DBInfo?

**Comment by zimulala** [comment]
PTAL @coocood @shenli

**Comment by coocood** [comment]
LGTM

**Pull request merged**